xbrowse_server/phenotips/utilities.py

```
@@ -0,0 +1,121 @@
+  uname=project_name+ '_view'
+  pwd=project_name+project_name
+  return uname,pwd
+
+
+
+#generates a new user in phenotips
+def add_new_user_to_phenotips(new_user_first_name, new_user_last_name,new_user_name,email_address,new_user_pwd):
+  #we need to put this password in a non-checkin file
+  admin_uname='Admin'
+  admin_pwd='admin'
+  headers={"Content-Type": "application/x-www-form-urlencoded"}
+  data={'parent':'XWiki.XWikiUsers'}
+  url = settings.PHENOPTIPS_HOST_NAME + '/rest/wikis/xwiki/spaces/XWiki/pages/' + new_user_name
+  do_authenticated_PUT(admin_uname,admin_pwd,url,data,headers) 
+  data={'className':'XWiki.XWikiUsers',
+        'property#first_name':new_user_first_name,
+        'property#last_name':new_user_last_name,
+        'property#email':email_address,
+        'property#password':new_user_pwd
+        }
+  url=settings.PHENOPTIPS_HOST_NAME + '/rest/wikis/xwiki/spaces/XWiki/pages/' + new_user_name + '/objects'
+  do_authenticated_POST(admin_uname,admin_pwd,url,data,headers)
+  
+  
+#adds a collaborator to an existing patient. Requires an existing collaborator username, patient_eid (PXXXX..)
+def add_read_only_collaborator_phenotips_patient(collaborator_username,patient_eid):
+  #we need to put this password in a non-checkin file
+  admin_uname='Admin'
+  admin_pwd='admin'
+  headers={"Content-Type": "application/x-www-form-urlencoded"}
+  data={'collaborator':'XWiki.' + collaborator_username,
+        'patient':patient_eid,
+        'accessLevel':'view',
+        'xaction':'update',
+        'submit':'Update'}
+  url = settings.PHENOPTIPS_HOST_NAME + '/bin/get/PhenoTips/PatientAccessRightsManagement?outputSyntax=plain'
+  do_authenticated_POST(admin_uname,admin_pwd,url,data,headers)
+  
+  
+#do a PUT call to phenotips
+def do_authenticated_PUT(uname,pwd,url,data,headers):
+  try:
+    request=requests.put(url,data=data,auth=(uname,pwd),headers=headers)
+  except Exception as e:
+    print 'error in do_authenticated_PUT:',e,
+    return e
```

I'm fine with catching and logging the request errors here, but again I think better to re-raise the same error or throw a new one, than return a value.